pkgs/build-support/setup-hooks/make-symlinks-relative.sh

```diff
@@ -0,0 +1,28 @@
+fixupOutputHooks+=(_makeSymlinksRelative)
+
+# For every symlink in $output that refers to another file in $output
+# ensure that the symlink is relative. This removes references to the output
+# has from the resulting store paths and thus the NAR files.
+_makeSymlinksRelative() {
+    local symlinkTarget
+
+    if [ -n "${dontRewriteSymlinks}" ]; then
+        return 0
+    fi
+
+    while IFS= read -r -d $'\0' f; do
+        symlinkTarget=$(readlink "$f")
+        if [[ "$symlinkTarget"/ != "$prefix"/* ]]; then
+            # skip this symlink as it doesn't point to $prefix
+            continue
+        fi
+
+        if [ ! -e "$symlinkTarget" ]; then
```

Comment:
I think this block (lines 20-22) should preceed the block above (lines 15-18) so sanity of external symlinks will be checked too.

Reply:
I don't think that is a good idea. A brief look at nixpkgs showed that a bunch of packages point symlinks at `/etc/$foo/` or to `/run/opengl-drivers` and those clearly do not exist in the build sandbox. Expecting every single of those packages to opt out of this is not really a great solution. This should be an incremental change and not a roadblock.

Reply:
it is only printing a warning.
symlinks to `/etc/$foo/` also deserve to be warned about, it is a potential problem with Nix-on-{Debian,Fedora,Gentoo,...} where `/etc/$foo/` might differ from that on NixOS